core/internal/api/send.go

```diff
@@ -21,10 +22,47 @@ func (client *clientImpl) Send(req *Request) (*http.Response, error) {
 	for headerKey, headerValue := range req.Headers {
 		retryableReq.Header.Set(headerKey, headerValue)
 	}
-	client.setClientHeaders(retryableReq)
-	client.setAuthHeaders(retryableReq)
 
-	resp, err := client.retryableHTTP.Do(retryableReq)
+	return client.sendToWandbBackend(retryableReq)
+}
+
+func (client *clientImpl) Do(req *http.Request) (*http.Response, error) {
```

Comment:
what do you think of having 2 client implementations, one for wandb end points and one general, maybe it will be more readable and we won't need to do this conditionals? we could also use a single `Do` function?

Reply:
discussed in standup: this is more of a guardrail, and we expect all requests to go to the W&B backend. I added logging to see if this is ever not true.